lib/screens/lock/lock_screen.dart

```diff
@@ -0,0 +1,241 @@
+                Text(
+                  'You will also receive TIME tokens that can be used for single asset staking. Read more about the use of TIME token on our wiki',
+                  style: MyStyles.whiteMediumTextStyle,
+                ), //TODO: add link
+                kBigSpacer,
+                Padding(
+                  padding: const EdgeInsets.only(left: 15),
+                  child: Text(
+                    'Balance: $balance',
+                    style: MyStyles.lightWhiteSmallTextStyle,
+                  ),
+                ),
+                SizedBox(height: 5),
+                TextFieldWithMax(
+                  controller: _textController,
+                  maxValue: balance,
+                ),
+                kSmallSpacer,
+                DarkButton(
+                  label: 'Show me the contract',
+                  onPressed: () {},
+                  labelStyle: MyStyles.whiteMediumTextStyle,
+                ),
+                kSmallSpacer,
+                _buildLockApproveButton(),
+                kMediumSpacer,
+                if (_lockState == ButtonStates.hasToApprove) _buildSteps(),
+                Spacer(),
+                if ((_lockState != ButtonStates.hasToApprove && _showToast) ||
+                    (_lockState == ButtonStates.pendingApproveDividedButton &&
+                        _showToast))
+                  _buildToast()
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  CountdownTimer _buildCountDownTimer() {
+    return CountdownTimer(
+      endTime: endTime,
+      widgetBuilder: (_, time) {
+        if (time == null) {
+          return Text(
+              'Current locking period still lasts 0 Days, 0 Hours, 0 Minutes and 0 Seconds.',
+              style: MyStyles.gradientMediumTextStyle);
+        } else {
+          return Text(
+            'Current locking period still lasts ${time?.days ?? 0} Days, ${time?.hours ?? 0} Hours, ${time?.min ?? 0} Minutes and ${time?.sec ?? 0} Seconds.',
+            style: MyStyles.gradientMediumTextStyle,
+          );
+        }
+      },
+    );
+  }
+
+  Widget _buildToast() {
+    if (_lockState != ButtonStates.pendingApproveDividedButton &&
+        _lockState != ButtonStates.pendingApproveMergedButton) {
+      return _buildTransactionSuccessToast();
+    } else {
+      return _buildTransactionPending();
+    }
+  }
+
+  Toast _buildTransactionSuccessToast() {
+    return Toast(
+      label: 'Successful',
+      color: Color(MyColors.ToastGreen),
+      onPressed: () {},
+      onClosed: () {
+        setState(() {
+          _showToast = false;
+        });
+      },
+    );
+  }
+
+  Toast _buildTransactionPending() {
+    return Toast(
+      label: 'Transaction Pending',
+      color: Color(MyColors.ToastGrey),
+      onPressed: () {},
+      onClosed: () {
+        setState(() {
+          _showToast = false;
+        });
+      },
+    );
+  }
+
+  Padding _buildSteps() {
```

Comment:
Ich denke hier würde ich die Methode von dem einen Screen einfach public machen (also den Unterstrich im Methodennamen weg und im State genau so, oder die Methode in das Widget statt in den WidgetState verschieben) und dann hier aufrufen.

- [ ] LockScreen.buildSteps(); oder LockScreenState.buildSteps();
oder du holst es in eine extra Datei. Würde aber kein extra Widget dafür nehmen denke ich, weil es fast nur aus konstanten und einfachen Widgets besteht.